### PR TITLE
Toolsmiths/fix default int property

### PR DESCRIPTION
**Pull request by tdctaz**
We don't want to try to set properties for readonly values don't exists in the force form

Also adds a workaround for corrupted local python pickle? - not sure if this is a good fix or not or some automatic clean up would have been better.

### CODE REVIEW

**Review by mariangemarcano — APPROVED** [review]

**Review by campos-ddc — APPROVED** [review]

File: master/buildbot/schedulers/forcesched.py [hunk 1]
```diff
@@ -121,7 +123,9 @@ def getFromKwargs(self, kwargs):
 
     def updateFromKwargs(self, properties, kwargs, **unused):
         """Primary entry point to turn 'kwargs' into 'properties'"""
-        properties[self.name] = self.getFromKwargs(kwargs)
+        value = self.getFromKwargs(kwargs)
+        if value is not None:
```
Comment (Mowinski) [inline master/buildbot/schedulers/forcesched.py]: It will be much more readable when you write inverse condition. I mean, exception (value is None) has own branch, but rest of code is executed in the main branch (flat).

So, my suggestion is that, rewrite it in this way:
```
if value is None:
    return
 properties[self.name] = value
```

File: master/buildbot/schedulers/forcesched.py [hunk 2]
```diff
@@ -95,6 +95,8 @@ def getFromKwargs(self, kwargs):
         if len(args) == 0:
             if self.required:
                 raise ValidationError("'%s' needs to be specified" % (self.label))
+            if self.readonly:
```
Comment (Mowinski) [inline master/buildbot/schedulers/forcesched.py]: Also, I am not sure, that is a correct place for this condition. By name `getFromKwargs` should return something but you've put there a core logic condition.

Let's imagine that, you are a user of this class. When you called `getFromKwargs` you supposed that this method returns some dict or list from kwargs, but when you set read-only property on True, this function returns nothing. Is a special corner case which is very ugly. 

Can we move this condition to another place?

**Review by Mowinski — COMMENTED** [review]
And the main aspect. I think that would be nice to write some unit tests for this feature. It'll show how to use this feature in real cases.

Also: could you change the target branch to release candidate branch?

File: master/buildbot/schedulers/forcesched.py (hunk 1 above)
Comment (tdctaz) [reply master/buildbot/schedulers/forcesched.py]: Sure

File: master/buildbot/schedulers/forcesched.py (hunk 2 above)
Comment (tdctaz) [reply master/buildbot/schedulers/forcesched.py]: I can move it to `updateFromKwargs`, this was where I first had it, but it means that we will duplicate the logic a bit for with what is already inside `def getFromKwargs(self, kwargs):` which I did not like.

I'll try to move it back and see if you like that better.

**Review by Mowinski — APPROVED** [review]
